Binary should include default config

**Describe the bug**
Currently, it reads config from `internal/utils/defaults.yaml` then `~/.carbonifer/config.yaml` then `./.carbonifer/config.yaml`. When user runs it from source (`go run main.go ... `) it's working well and finds `internal/utils/defaults.yaml`, but when we run the binary from elsewhere, it's does not find the `defaults.yaml` config, and fails unless there is a `~/.carbonifer/config.yaml`:

**To Reproduce**
Steps to reproduce the behavior:
```
$ go install go install github.com/carboniferio/carbonifer@latest
$ cd path/to/some/terraform/project
$ carbonifer plan .
```
Then we get this error message: 

```
PANI[0000] Config File "config" Not Found in "[$HOME/.carbonifer /etc/carbonifer /path/to/some/terraform/project/.carbonifer]" 
panic: (*logrus.Entry) 0x14000197ab0

goroutine 1 [running]:
github.com/sirupsen/logrus.(*Entry).log(0x14000197a40, 0x0, {0x140000ee420, 0xa2})
        /Users/olivier/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/entry.go:260 +0x318
github.com/sirupsen/logrus.(*Entry).Log(0x14000197a40, 0x0, {0x1400011dc80, 0x1, 0x1})
        /Users/olivier/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/entry.go:304 +0x7c
github.com/sirupsen/logrus.(*Logger).Log(0x140000ca000, 0x0, {0x1400011dc80, 0x1, 0x1})
        /Users/olivier/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/logger.go:204 +0x70
github.com/sirupsen/logrus.(*Logger).Panic(...)
        /Users/olivier/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/logger.go:253
github.com/sirupsen/logrus.Panic(...)
        /Users/olivier/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/exported.go:129
github.com/carboniferio/carbonifer/cmd.initConfig()
        /Users/olivier/go/pkg/mod/github.com/carboniferio/carbonifer@v0.2.0/cmd/root.go:88 +0x288
```

**Expected behavior**
Have default config value included in the binary and should not fail if no config file exists.

--- a/cmd/root.go
+++ b/cmd/root.go
@@ -17,10 +17,8 @@ package cmd
 
 import (
 	"os"
-	"path"
 
 	"github.com/carboniferio/carbonifer/internal/utils"
-	"github.com/heirko/go-contrib/logrusHelper"
 	"github.com/sirupsen/logrus"
 	log "github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
@@ -64,42 +62,16 @@ func init() {
 
 // initConfig reads in config file and ENV variables if set.
 func initConfig() {
-	if cfgFile != "" {
-		// Use config file from the flag.
-		viper.SetConfigFile(cfgFile)
-	} else {
-		// Find home directory.
-		home, err := os.UserHomeDir()
-		cobra.CheckErr(err)
+	utils.InitConfig(cfgFile)
 
-		viper.AddConfigPath(path.Join(home, ".carbonifer"))
-		viper.AddConfigPath("/etc/carbonifer/")
-		viper.AddConfigPath("./.carbonifer")
-		if viper.ConfigFileUsed() == "" {
-			viper.SetConfigType("yaml")
-			viper.SetConfigName("config")
-		}
-	}
-
-	viper.AutomaticEnv() // read in environment variables that match
-
-	// If a config file is found, read it in.
-	if err := viper.ReadInConfig(); err != nil {
-		log.Panic(err)
-	}
-
-	// Setup Logrus
-	var logrusConfig = logrusHelper.UnmarshalConfiguration(viper.GetViper().Sub("log")) // Unmarshal configuration from Viper
-	err := logrusHelper.SetConfig(log.StandardLogger(), logrusConfig)
+	// Set working directory
+	currentDir, err := os.Getwd()
 	if err != nil {
-		log.Panic(err)
+		log.Fatal(err)
 	}
+	viper.SetDefault("workdir", currentDir)
 
-	if viper.ConfigFileUsed() != "" {
-		log.Infof("Using config file: %v", viper.ConfigFileUsed())
-	}
-
-	// Set log level
+	// Set log level from command flags
 	info, _ := RootCmd.Flags().GetBool("info")
 	debug, _ := RootCmd.Flags().GetBool("debug")
 	if info {
@@ -109,14 +81,6 @@ func initConfig() {
 	} else {
 		viper.SetDefault("log.level", "warning")
 	}
-
-	// Viper default values
-	currentDir, err := os.Getwd()
-	if err != nil {
-		log.Fatal(err)
-	}
-	viper.SetDefault("workdir", currentDir)
-	utils.LoadViperDefaults()
 
 	// Bind Viper and Cobra flags
 	if err := viper.BindPFlag("out.format", RootCmd.PersistentFlags().Lookup("format")); err != nil {

--- a/internal/testutils/testutils.go
+++ b/internal/testutils/testutils.go
@@ -1,14 +1,12 @@
 package testutils
 
 import (
-	"log"
 	"os"
 	"path"
 	"runtime"
 	"testing"
 
 	"github.com/carboniferio/carbonifer/internal/utils"
-	"github.com/spf13/viper"
 )
 
 var RootDir string
@@ -20,13 +18,7 @@ func init() {
 	if err != nil {
 		panic(err)
 	}
-	utils.LoadViperDefaults()
-	viper.AddConfigPath(path.Join(RootDir, "test/config"))
-	viper.SetConfigName("default_conf")
-	viper.SetConfigType("yaml")
-	if err := viper.ReadInConfig(); err != nil {
-		log.Fatal(err)
-	}
+	utils.InitConfig(path.Join(RootDir, "test/config/default_conf.yaml"))
 
 	// Set fake GCP auth
 	os.Setenv("GOOGLE_OAUTH_ACCESS_TOKEN", "foo")

--- a/internal/utils/config.go
+++ b/internal/utils/config.go
@@ -1,29 +1,35 @@
 package utils
 
 import (
+	_ "embed"
+	"io"
 	"os"
+	"path"
+	"path/filepath"
 	"sort"
 
 	"github.com/carboniferio/carbonifer/internal/estimate/estimation"
+	"github.com/heirko/go-contrib/logrusHelper"
 	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
 
 	"gopkg.in/yaml.v3"
 )
 
-func LoadViperDefaults() {
-	defaultConfigFile, err := os.ReadFile("internal/utils/defaults.yaml")
-	if err != nil {
-		log.Fatal(err)
-	}
+func InitConfig(configFilePath string) {
+	initViper(configFilePath)
+	initLogger()
+	checkDataConfig()
+}
 
-	if err != nil {
-		log.Fatal(err)
-	}
+//go:embed defaults.yaml
+var defaultConfigFile []byte
 
-	defaults := make(map[string]interface{})
+func loadViperDefaults() {
+	var defaults map[string]interface{}
 
-	err = yaml.Unmarshal(defaultConfigFile, &defaults)
+	err := yaml.Unmarshal(defaultConfigFile, &defaults)
 	if err != nil {
 		log.Fatal(err)
 	}
@@ -34,6 +40,72 @@ func LoadViperDefaults() {
 	settings := viper.AllSettings()
 
 	log.Debug(settings)
+}
+
+func initViper(configFilePath string) {
+	loadViperDefaults()
+	if configFilePath != "" {
+		// Use config file from the flag.
+		viper.SetConfigFile(configFilePath)
+	} else {
+		// Find home directory.
+		home, err := os.UserHomeDir()
+		cobra.CheckErr(err)
+
+		viper.AddConfigPath(path.Join(home, ".carbonifer"))
+		viper.AddConfigPath("/etc/carbonifer/")
+		viper.AddConfigPath("./.carbonifer")
+		if viper.ConfigFileUsed() == "" {
+			viper.SetConfigType("yaml")
+			viper.SetConfigName("config")
+		}
+	}
+
+	viper.AutomaticEnv() // read in environment variables that match
+
+	// If a config file is found, read it in.
+	if err := viper.ReadInConfig(); err != nil {
+		if _, ok := err.(viper.ConfigFileNotFoundError); !ok {
+			log.Panic(err)
+		}
+	}
+
+	if viper.ConfigFileUsed() != "" {
+		log.Infof("Using config file: %v", viper.ConfigFileUsed())
+	}
+}
+
+func initLogger() {
+	// Setup Logrus
+	logConf := viper.GetViper().Sub("log")
+	if logConf != nil {
+		var logrusConfig = logrusHelper.UnmarshalConfiguration(logConf) // Unmarshal configuration from Viper
+		err := logrusHelper.SetConfig(log.StandardLogger(), logrusConfig)
+		if err != nil {
+			log.Panic(err)
+		}
+	}
+}
+
+func checkDataConfig() {
+	dataPath := viper.GetString("data.path")
+	if dataPath == "" {
+		log.Fatalf("Data directory is not set (\"data.path\")")
+	}
+	path, err := filepath.Abs(dataPath)
+	if err != nil {
+		log.Fatal(err)
+	}
+	f, err := os.Open(dataPath)
+	if err != nil {
+		log.Fatalf("Cannot read data directory \"%v\": %v", path, err)
+	}
+	defer f.Close()
+
+	_, err = f.Readdirnames(1)
+	if err == io.EOF {
+		log.Fatalf("Empty data directory \"%v\": %v", path, err)
+	}
 }
 
 func SortEstimations(resources *[]estimation.EstimationResource) {

--- a/internal/utils/defaults.yaml
+++ b/internal/utils/defaults.yaml
@@ -20,3 +20,5 @@ provider:
     disk:
       size: 500
       type: pd-standard
+log:
+  level : "warn"

